### PR TITLE
Exploding messages: misc 2

### PR DESCRIPTION
**Pull request by buoyad**
- [x] Fix repeating ashes image on android
  - Spent some time trying to apply [this](https://github.com/facebook/react-native/commit/0459e4ffaadb161598ce1a5b14c08d49a9257c9c) with no luck
- [x] Handle collapsing ashes on scrolling around on desktop (kind of dirty, TODO'd a better way. it would be easier to store the heights locally in the component and not have them collapse until the app restarts.)
- [x] Various styling cleanup (~exploding popup on mobile~, message wrapper, ~set-explode-time popup~)
- [x] Look into synchronizing timers in message popup & `ExplodingMeta`
- [x] Add a lock on chat input so the exploding mode never changes while user is typing (maybe a separate PR)
- [x] ~Sort out order of BOOM! icon disappearing & `EXPLODED` appearing~ moved to separate ticket
- [x] Figure out why exploding meta stops counting sometimes when message popup is open (doesn't seem to happen anymore)
- [x] Add avatar to normal message menu
- [x] Handle erroneous inserts into gregor w/r/t `seenExploding` and `newExploding`

### CODE REVIEW

**Comment by buoyad** [comment]
@keybase/react-hackers this is ready for review!

File: shared/actions/gregor.js [hunk 1]
```diff
@@ -178,10 +178,38 @@ function handleConvExplodingModes(items: Array<Types.NonNullGregorItem>) {
...
+  if (
+    oldestSeenExploding &&
+    oldest &&
+    oldest.md.ctime - oldestSeenExploding.md.ctime > ChatConstants.newExplodingGregorOffset
+  ) {
+    newExplodings.push(oldest)
+  }
+  ret.push(...newExplodings)
+
+  console.log('dannydebug', ret)
```
Comment (chrisnojima) [inline shared/actions/gregor.js]: 🔥

File: shared/chat/conversation/input-area/normal/container.js [hunk 2]
```diff
@@ -105,7 +107,14 @@ const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps): Props => ({
   sendTyping: (typing: boolean) => {
     dispatchProps._sendTyping(stateProps.conversationIDKey, typing)
   },
-  setUnsentText: (text: string) => setUnsentText(stateProps.conversationIDKey, text),
+  setUnsentText: (text: string) => {
+    if (text.length > 0) {
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/container.js]: lets avoid duplicating code to just have a single var difference, prefer
```
const lock = text.length > 0
dispatchProps.onSetExplodingModeLock(stateProps.conversationIDKey, lock)
```

File: shared/chat/conversation/messages/index.js [hunk 3]
```diff
@@ -103,6 +103,14 @@ const mergeProps = (stateProps, dispatchProps, ownProps) => ({
 export default compose(
   connect(mapStateToProps, () => ({}), mergeProps),
   lifecycle({
+    componentDidMount() {
+      if (this.props._measure && this.props.message.exploded) {
+        // This fixes an issue with exploding messages not retaining
+        // their height after being exploded and scrolling around.
+        // TODO figure out a real less perf heavy way to do this
+        this.props._measure()
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/index.js]: lets chat about this

File: shared/util/second-timer.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// @flow
+
```
Comment (chrisnojima) [inline shared/util/second-timer.js]: should add a comment telling us what this is

File: shared/util/second-timer.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        clearInterval(this.intervalID)
+        this.intervalID = null
+      }
+      return true
+    }
+    return false
+  }
+
+  loop = () => {
+    this.refs.forEach(r => setImmediate(r.fn))
```
Comment (chrisnojima) [inline shared/util/second-timer.js]: why setImmediate

File: shared/util/second-timer.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  }
+}
```
Comment (chrisnojima) [inline shared/util/second-timer.js]: you might want to add a localDebug flag like the outstandingRPC one where we can log if there's outstanding listeners to this thing

File: shared/util/second-timer.js (hunk 5 above)
Comment (buoyad) [reply shared/util/second-timer.js]: I added this debugging some triggers that weren't in sync. Doesn't seem necessary now, removed.

File: shared/util/second-timer.js [hunk 7]
```diff
@@ -21,6 +23,25 @@ class Ticker {
...
+  }
+
+  _setupDebugging = () => {
+    if (!__DEV__) {
+      return
+    }
+
+    if (printOutstandingTimerListeners) {
+      // Print any listeners to this periodically
+      setInterval(() => {
```
Comment (chrisnojima) [inline shared/util/second-timer.js]: you can actually do this in your one timer

Comment (buoyad) [reply shared/util/second-timer.js]: done :+1:

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.js.flow [hunk 8]
```diff
@@ -5,6 +5,7 @@ import type {StylesCrossPlatform} from '../../../../../styles'
 export type Props = {
   children?: React.Node,
   explodedBy: ?string,
+  exploding: boolean,
```
Comment (buoyad) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.js.flow]: @chrisnojima `messageHeights` had 1500+ keys after scrolling up for a minute in chat if we don't limit it to exploding messages.

**Comment by buoyad** [comment]
@chrisnojima I also added cancelling edits if the message being edited explodes

**Comment by junderw** [comment]
Huzzah!

**Pull request merged**